5.0/en/0x12-V3-Session-management.md

```diff
@@ -91,10 +90,88 @@ Attackers start a half-open attack by attempting to lock, reset, or recover a cr
 | :---: | :--- | :---: | :---: | :---: | :---: | :---: |
 | **3.7.1** | [MODIFIED] Verify the application requires re-authentication or secondary factor verification before allowing any sensitive transactions or account modifications. | ✓ | ✓ | ✓ | 306 | |
 
+
+
+## V3.8 OAuth 2.0 Protocol
+
+This section describes the best current security practices for OAuth 2.0 as derived from its RFC. OAuth became the standard for API protection and the basis for federated login using OpenID Connect. OpenID Connect 1.0 is a simple identity layer on top of the OAuth 2.0 protocol. It enables clients to verify the identity of the end-user based on the authentication performed by an authorization server, as well as to obtain basic profile information about the end-user in an interoperable and REST-like manner. 
+
+## Terminology: 
+
+* `Access tokens` --- provides an abstraction, replacing different authorization constructs (e.g., username and password, assertion) for a single token understood by the resource server. This abstraction enables issuing access tokens valid for a short time period, as well as removing the resource server's need to understand a wide range of authentication schemes. 
+* `Refresh tokens` --- are credentials used to obtain access tokens. These are issued to the client by the authorization server and are used to obtain a new access token when the current access token becomes invalid or expires, or to obtain additional access tokens with identical or narrower scope (access tokens may have a shorter lifetime and fewer permissions than authorized by the resource owner). 
+* `Client` --- generally refers to an application making protected resource requests on behalf of the resource owner and with its authorization. The term "client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices). 
+* `Authorization Server (AS)` --- refers to the server issuing access tokens to the client after successfully authenticating the resource owner and obtaining authorization.
+* `Resource Owner (RO)` --- refers to an entity capable of granting access to a protected resource. When the resource owner is a person, it is referred to as an end-user. 
+* `Resource Server (RS)` --- refers to the server hosting the protected resources, capable of accepting and responding to protected resource requests using access tokens. 
+
+
+## OAuth 2.0 Essential Basics
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **3.8.1** | [ADDED] Verify that Clients do not expose URLs that forward the user's browser to arbitrary URIs obtained from a query parameter ("open redirectors") which can enable exfiltration of authorization codes and access tokens.  | ✓ | ✓ | ✓ |
```

Comment:
I think you are right, this is somehow covered in 5.1.5, but we are just emphasizing here for OAuth-related security which I think makes it better that it mentions to prevent exfiltration of authorization codes and access tokens so that developers/engineers understand why it is important not to use URLs. I think we can keep this unless you strongly think otherwise?

Reply:
I think it is already covered, I don't really want to double up as there are already so many extra requirements. I think let's remove it. Maybe we should have a note at the start of other important requirements that are not OAuth specific but often feature in OAuth related vulnerabilities?

Reply:
Ah good point, I'll just add that OAuth specific info in 5.1.5 as a note, probably in the intro paragraph of that subsection.

Reply:
Ok thanks